helpers/compile/plugins/fill-plugin/fillers/buffer-small.ts

```diff
@@ -0,0 +1,704 @@
+    return BufferClass.from(size)
+  }
+
+  static allocUnsafeSlow(size: number) {
+    return BufferClass.from(size)
+  }
+
+  static isBuffer(arg: any): arg is BufferClass {
+    return arg && !!arg._isBuffer
+  }
+
+  static byteLength(string: unknown, encoding: Encoding = 'utf8') {
+    if (typeof string === 'string') return stringToBuffer(string, encoding).byteLength
+    if (string && string['byteLength']) return string['byteLength']
+
+    const e = new TypeError('The "string" argument must be of type string or an instance of Buffer or ArrayBuffer.')
+    e['code'] = 'ERR_INVALID_ARG_TYPE'
+    throw e
+  }
+
+  static isEncoding(encoding: string): encoding is Encoding {
+    return Encodings.includes(encoding as any)
+  }
+
+  static compare(buff1: Uint8Array, buff2: Uint8Array) {
+    assertUint8Array(buff1, 'buff1')
+    assertUint8Array(buff2, 'buff2')
+
+    for (let i = 0; i < buff1.length; i++) {
+      if (buff1[i] < buff2[i]) return -1
+      if (buff1[i] > buff2[i]) return 1
+    }
+
+    return buff1.length === buff2.length ? 0 : buff1.length > buff2.length ? 1 : -1
+  }
+
+  static from(value: unknown, encoding: any = 'utf8'): BufferClass {
+    if (value && typeof value === 'object' && value['type'] === 'Buffer') {
+      return new BufferClass(value['data'])
+    }
+
+    if (typeof value === 'number') {
+      return new BufferClass(new Uint8Array(value))
+    }
+
+    if (typeof value === 'string') {
+      return stringToBuffer(value, encoding)
+    }
+
+    if (ArrayBuffer.isView(value)) {
+      const { byteOffset, byteLength, buffer } = value
+
+      if ('map' in value && typeof value.map === 'function') {
+        return new BufferClass(
+          value.map((v) => v % 256),
+          byteOffset,
+          byteLength,
+        )
+      }
+
+      return new BufferClass(buffer, byteOffset, byteLength)
+    }
+
+    if (value && typeof value === 'object' && ('length' in value || 'byteLength' in value || 'buffer' in value)) {
+      return new BufferClass(value as ArrayLike<number>)
+    }
+
+    throw new TypeError('First argument must be a string, Buffer, ArrayBuffer, Array, or array-like object.')
+  }
+
+  static concat(list: readonly Uint8Array[], totalLength?: number) {
+    if (list.length === 0) return BufferClass.alloc(0)
+
+    const concat = ([] as number[]).concat(...list.map((item) => [...item]))
+    const result = BufferClass.alloc(totalLength !== undefined ? totalLength : concat.length)
+
+    result.set(totalLength !== undefined ? concat.slice(0, totalLength) : concat)
+
+    return result
+  }
+
+  slice(start = 0, end = this.length) {
+    return this.subarray(start, end)
+  }
+
+  subarray(start = 0, end = this.length) {
+    return Object.setPrototypeOf(super.subarray(start, end), BufferClass.prototype) as BufferClass
+  }
+
+  reverse() {
+    super.reverse()
+
+    return this
+  }
+
+  readIntBE(offset: number, byteLength: number) {
+    assertNumber(offset, 'offset')
```

Comment:
Does not need to change, but do you thing we can get a little bit of size win if we pull all those strings into constants? Var name can be than minified.

Reply:
I checked and turns out not so much: Since the gzip size is what matters and these are put into a dictionary, putting them into a variable produced minimal gains unfortunately.